Source/WebKit/NetworkProcess/storage/NetworkStorageManager.cpp

```diff
@@ -799,11 +799,34 @@ void NetworkStorageManager::didIncreaseQuota(WebCore::ClientOrigin&& origin, Quo
     });
 }
 
+void NetworkStorageManager::isFileAllowedForConnection(IPC::Connection::UniqueID connectionID, const String& path, CompletionHandler<void(bool)>&& completionHandler) const
+{
+    ASSERT(RunLoop::isMain());
+    ASSERT(!m_closed);
+
+    m_queue->dispatch([this, protectedThis = Ref { *this }, connectionID, completionHandler = WTFMove(completionHandler), path = path.isolatedCopy()]() mutable {
```

Comment:
Why is this on a background queue?  It's just doing string operations, not file I/O, right?

Reply:
`m_temporaryBlobPathsByConnection` may only be accessed on the background queue. 

`WTF_GUARDED_BY_CAPABILITY(workQueue());`